src/function/functions.cpp

```diff
@@ -0,0 +1,35 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// functions.cpp
+//
+// Identification: src/function/functions.cpp
+//
+// Copyright (c) 2015-2017, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+
+#include "function/functions.h"
+
+namespace peloton {
+namespace function {
+
+std::unordered_map<std::string, BuiltInFuncType>
```

Comment:
I guess having a map works for now. You will be putting the function pointers into the catalog in the future changes, right?

Reply:
Andy suggests not to put function pointers into the catalog, just to use a simple switch statement here. I think using a map here is the same as a a switch statement, ans is easier to use.